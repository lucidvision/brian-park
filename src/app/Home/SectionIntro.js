import React from 'react'
import styled from 'styled-components'
import { HeadingPrimary, ImageCircle, Paragraph } from 'components'
import brian from 'assets/brian.jpg'

const StyledSection = styled.section`
  text-align: center;
`

export default function SectionIntro (props) {
  return (
    <StyledSection>
      <ImageCircle imagePath={brian} />
      <HeadingPrimary text={'Hi, I\'m Brian'} />
      <Paragraph primary text='I have a passion for bringing products to market - from product management to software engineering.' />
    </StyledSection>
  )
}
