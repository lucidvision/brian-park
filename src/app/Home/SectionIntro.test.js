import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import SectionIntro from './SectionIntro'

Enzyme.configure({adapter: new Adapter()})

describe('SectionIntro', () => {
  const sectionIntro = shallow(<SectionIntro />)

  it('renders correctly', () => {
    expect(sectionIntro).toMatchSnapshot()
  })

  it('renders the video component', () => {
    expect(sectionIntro.find('VideoCode').exists()).toBe(true)
  })

  it('renders a primary heading', () => {
    expect(sectionIntro.find('Styled(HeadingPrimary)').at(0).props().text).toEqual('Brian Park')
  })

  it('renders a sub heading', () => {
    expect(sectionIntro.find('Styled(HeadingPrimary)').at(1).props().text).toEqual('Bringing products to market')
  })
})
