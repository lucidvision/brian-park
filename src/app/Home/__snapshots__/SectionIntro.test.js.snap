// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionIntro renders correctly 1`] = `
<styled.section>
  <styled.div>
    <VideoCode />
    <styled.div>
      <Styled(HeadingPrimary)
        color="#fff"
        text="Brian Park"
      />
      <Styled(HeadingPrimary)
        color="#fff"
        size="2.2rem"
        text="Bringing products to market"
      />
    </styled.div>
    <a
      href="#skills"
    >
      <styled.i
        className="fas fa-angle-down"
      />
    </a>
  </styled.div>
</styled.section>
`;
